File: app/src/main/java/igrek/todotree/mock/MockedGUI.java
package igrek.todotree.mock;


import android.support.v7.app.AppCompatActivity;
import android.view.View;

import java.util.List;
import java.util.Set;

import igrek.todotree.model.treeitem.AbstractTreeItem;
import igrek.todotree.ui.GUI;

public class MockedGUI extends GUI {
	
	public MockedGUI(AppCompatActivity activity) {
		super(activity);
	}
	
	@Override
	public void lazyInit() {
	}
	
	@Override
	public void showItemsList(AbstractTreeItem currentItem) {
		updateItemsList(currentItem, null, null);
	}
	
	@Override
	public void showEditItemPanel(AbstractTreeItem item, AbstractTreeItem parent) {
	}
	
	@Override
	public View showExitScreen() {
		return null;
	}
	
	@Override
	public void updateItemsList(AbstractTreeItem currentItem, List<AbstractTreeItem> items, Set<Integer> selectedPositions) {
	}
	
	@Override
	public void scrollToItem(int itemIndex) {
	}
	
	@Override
	public void scrollToItem(Integer y, int itemIndex) {
	}
	
	@Override
	public void scrollToPosition(int y) {
	}
	
	@Override
	public void scrollToBottom() {
	}
	
	@Override
	public void hideSoftKeyboard() {
	}
	
	@Override
	public boolean editItemBackClicked() {
		throw new UnsupportedOperationException();
	}
	
	@Override
	public void setTitle(String title) {
	}
	
	@Override
	public Integer getCurrentScrollPos() {
		return 0;
	}
	
	@Override
	public void requestSaveEditedItem() {
		throw new UnsupportedOperationException();
	}
	
	@Override
	public void rotateScreen() {
		throw new UnsupportedOperationException();
	}
	
	@Override
	public void quickInsertRange() {
		throw new UnsupportedOperationException();
	}
}

File: app/src/test/java/igrek/todotree/mock/MockedGUI.kt
package igrek.todotree.mock

import android.view.View
import androidx.appcompat.app.AppCompatActivity
import igrek.todotree.domain.treeitem.AbstractTreeItem
import igrek.todotree.ui.GUI

class MockedGUI(activity: AppCompatActivity?) : GUI(activity) {
    override fun lazyInit() {}
    override fun showItemsList(currentItem: AbstractTreeItem) {
        updateItemsList(currentItem, null, null)
    }

    override fun showEditItemPanel(item: AbstractTreeItem?, parent: AbstractTreeItem?) {}
    override fun showExitScreen(): View? {
        return null
    }

    override fun updateItemsList(currentItem: AbstractTreeItem, items: List<AbstractTreeItem?>?, selectedPositions: Set<Int?>?) {}
    override fun scrollToItem(itemIndex: Int) {}
    override fun scrollToItem(y: Int?, itemIndex: Int) {}
    override fun scrollToPosition(y: Int) {}
    override fun scrollToBottom() {}
    override fun hideSoftKeyboard() {}
    override fun editItemBackClicked(): Boolean {
        throw UnsupportedOperationException()
    }

    override fun setTitle(title: String?) {}
    override val currentScrollPos: Int
        get() = 0

    override fun requestSaveEditedItem() {
        throw UnsupportedOperationException()
    }

    override fun rotateScreen() {
        throw UnsupportedOperationException()
    }

    override fun quickInsertRange() {
        throw UnsupportedOperationException()
    }
}